sympy/matrices/expressions/dotproduct.py

```diff
@@ -0,0 +1,56 @@
+from __future__ import print_function, division
+
+from sympy.core import Basic, sympify
+from sympy.matrices.expressions.matexpr import (MatrixExpr)
+
+
+class DotProduct(MatrixExpr):
+    """
+        Dot Product of vector matrices
+    """
+
+    def __new__(cls, *args):
+        args = list(map(sympify, args))
+        if validate(*args):
+            return Basic.__new__(cls, *args)
+
+    @property
+    def args(self):
+        return self.args
```

Comment:
This will give infinite recursion. I don't think you need to define args at all. It will be taken care of from subclassing Basic.